### PR TITLE
Implement get_expected_withdrawals

### PR DESCRIPTION
**Pull request by potuz**
This PR implements `get_expected_withdrawals` as in the specs. It produces the next withdrawals that should be packed in an execution payload. 

This PR implements some helpers as pure functions when they are already implemented as `ReadOnlyValidator` these should be removed in a future PR when the implementation is more clear

https://github.com/prysmaticlabs/prysm/issues/11619

### CODE REVIEW

File: beacon-chain/state/state-native/getters_withdrawal.go [hunk 1]
```diff
@@ -29,3 +35,80 @@ func (b *BeaconState) LastWithdrawalValidatorIndex() (types.ValidatorIndex, erro
 
 	return b.lastWithdrawalValidatorIndex, nil
 }
+
+// hasETH1WithdrawalCredential returns whether the validator has an ETH1
+// Withdrawal prefix. It assumes that the caller has a lock on the state
+func hasETH1WithdrawalCredential(val *ethpb.Validator) bool {
```
Comment (rkapka) [inline beacon-chain/state/state-native/getters_withdrawal.go:41]: Can you move all unexported functions to the bottom?

File: beacon-chain/state/state-native/getters_withdrawal_test.go [hunk 2]
```diff
@@ -36,3 +39,272 @@ func TestLastWithdrawalValidatorIndex(t *testing.T) {
...
+		require.Equal(t, params.BeaconConfig().MaxWithdrawalsPerPayload, uint64(len(expected)))
+		withdrawal := &enginev1.Withdrawal{
+			WithdrawalIndex:  0,
+			ValidatorIndex:   1,
+			ExecutionAddress: s.validators[0].WithdrawalCredentials[12:],
+			Amount:           params.BeaconConfig().MaxEffectiveBalance + 1,
+		}
+		require.DeepEqual(t, withdrawal, expected[0])
+	})
+
```
Comment (rkapka) [inline beacon-chain/state/state-native/getters_withdrawal_test.go:309]:
```suggestion
```

File: beacon-chain/state/state-native/getters_withdrawal_test.go [hunk 3]
```diff
@@ -36,3 +39,272 @@ func TestLastWithdrawalValidatorIndex(t *testing.T) {
...
+		}
+		withdrawalPartial := &enginev1.Withdrawal{
+			WithdrawalIndex:  0,
+			ValidatorIndex:   3,
+			ExecutionAddress: s.validators[3].WithdrawalCredentials[12:],
+			Amount:           params.BeaconConfig().MinDepositAmount,
+		}
+		require.DeepEqual(t, withdrawalPartial, expected[0])
+		require.DeepEqual(t, withdrawalFull, expected[1])
+
```
Comment (rkapka) [inline beacon-chain/state/state-native/getters_withdrawal_test.go:226]:
```suggestion
```

File: beacon-chain/state/state-native/getters_withdrawal_test.go [hunk 4]
```diff
@@ -36,3 +39,272 @@ func TestLastWithdrawalValidatorIndex(t *testing.T) {
...
+	creds = []byte{params.BeaconConfig().ETH1AddressWithdrawalPrefixByte, 0xCC}
+	v, err = NewValidator(&ethpb.Validator{
+		WithdrawalCredentials: creds,
+		EffectiveBalance:      params.BeaconConfig().MaxEffectiveBalance,
+	})
+	require.NoError(t, err)
+	require.Equal(t, true, v.IsPartiallyWithdrawable(params.BeaconConfig().MaxEffectiveBalance+1))
+}
+
+func TestExpectedWithdrawals(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (potuz) [reply beacon-chain/state/state-native/getters_withdrawal_test.go:113]: All tests do this.

**Review by rkapka — APPROVED** [review]

**Pull request merged**